architecture/builds.adoc

```diff
@@ -0,0 +1,52 @@
+= Builds
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+A build is a process of transforming input parameters into resulting object.
```

Comment:
transforming inputs into a runnable object?  (the parameters indicate what the inputs are, but they are not the actual inputs to the build, ultimately, and the resulting object is more specific than just "some object")

bear in mind of course that with custombuild the output does not have to be an image, it could just update some health check information on a website or something.